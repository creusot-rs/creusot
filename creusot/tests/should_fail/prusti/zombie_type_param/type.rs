extern crate creusot_contracts;
use creusot_contracts::prusti_prelude::*;

pub trait Trait {}

impl<T> Trait for T {}
enum MyOp<X: Trait> {
    None,
    Some(X),
}

#[ensures(MyOp::Some(x) != MyOp::None)]
pub fn test_op(x: Box<u32>) {}
