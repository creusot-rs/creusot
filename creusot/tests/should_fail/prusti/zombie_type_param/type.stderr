error: Unsatisfied bound creusot_contracts::prusti::Zombie<std::boxed::Box<u32>>: Trait
  --> type.rs:12:11
   |
12 | #[ensures(MyOp::Some(x) != MyOp::None)]
   |           ^^^^^^^^^^^^^

error: aborting due to 1 previous error

